stable_baselines/common/env_checker.py

```diff
@@ -0,0 +1,218 @@
+    :param headless: (bool) Whether to disable render modes
+        that require a graphical interface. False by default.
+    """
+    render_modes = env.metadata.get('render.modes')
+    if render_modes is None:
+        if warn:
+            warnings.warn("No render modes was declared in the environment "
+                          " (env.metadata['render.modes'] is None or not defined), "
+                          "you may have trouble when calling `.render()`")
+
+    else:
+        # Don't check render mode that require a
+        # graphical interface (useful for CI)
+        if headless and 'human' in render_modes:
+            render_modes.remove('human')
+        # Check all declared render modes
+        for render_mode in render_modes:
+            env.render(mode=render_mode)
+        env.close()
+
+
+def check_env(env: gym.Env, warn=True, skip_render_check=True):
+    """
+    Check that an environment follows Gym API.
+    This is particularly useful when using a custom environment.
+    Please take a look at https://github.com/openai/gym/blob/master/gym/core.py
+    for more information about the API.
+
+    It also optionally check that the environment is compatible with Stable-Baselines.
+
+    :param env: (gym.Env) The Gym environment that will be checked
+    :param warn: (bool) Whether to output additional warnings
+        mainly related to the interaction with Stable Baselines
+    :param skip_render_check: (bool) Whether to skip the checks for the render method.
+        True by default (useful for the CI)
+    """
+    assert isinstance(env, gym.Env), ("You environment must inherit from gym.Env class "
+                                      " cf https://github.com/openai/gym/blob/master/gym/core.py")
+
+    # ============= Check the spaces (observation and action) ================
+    _check_spaces(env)
+
+    # Define aliases for convenienve
+    observation_space = env.observation_space
+    action_space = env.action_space
+
+    # Warn the user if needed.
+    # A warning means that the environment may run but not work properly with Stable Baselines algorithms
+    if warn:
+        _check_unsupported_obs_spaces(env, observation_space)
+
+        # If image, check the low and high values, the type and the number of channels
+        # and the shape (minimal value)
+        if isinstance(observation_space, spaces.Box) and len(observation_space.shape) == 3:
+            _check_image_input(observation_space)
+
+        if isinstance(observation_space, spaces.Box) and len(observation_space.shape) not in [1, 3]:
+            warnings.warn("Your observation has an unconventional shape (neither an image, nor a 1D vector). "
+                          "We recommend you to flatten the observation "
+                          "to have only a 1D vector")
+
+        # Check for the action space, it may lead to hard-to-debug issues
+        if isinstance(action_space, spaces.Box) and \
```

Comment:
I prefer implicit line wrapping using parentheses similar to http://google.github.io/styleguide/pyguide.html#32-line-length however I've seen both styles in Stable Baselines. We should probably decide on one and stick to it though.

Reply:
done ;)